### PR TITLE
Use FileReader for TLS CA and Registry CA upload

### LINKED ISSUE #12: Use Web APIs' FileReader for certification files upload
**User Statement:**
As a user, I want to be able to use a VCH with custom server/client certificates so authentication could be properly performed

**Details:**
Web API's FileReader allows reading a file from the client, which we could use to send certificates along with other payload information to the vic-machine API endpoint.

**Acceptance Criteria:**
TBD

### PR DESCRIPTION
**Pull request by jooskim**
Fixes #12

### CODE REVIEW

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.ts [hunk 1]
```diff
@@ -237,4 +255,49 @@ export class SecurityComponent implements OnInit {
...
+                    });
+                }
+            };
+        };
+
+        // since input is without the 'multiple' attribute we are sure that
+        // only one entry will be available under FileList
+        const fileInstance: File = fileList[0];
+        if (!fileInstance) {
+            console.error('FileReader failed to load the file!');
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.ts]: We should probably output this as an actual UI error to the user if there is a failure.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.ts [hunk 2]
```diff
@@ -30,6 +30,8 @@ export class SecurityComponent implements OnInit {
     public formErrMessage = '';
     public inAdvancedMode = false;
     public signpostOpenState = false;
+    private tlsCaContents: any[] = [];
+    private registryCaContents: any[] = [];
     @Input() vchName: string;
 
     // tmp
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.ts]: Why is this tmp?

Also line :66 there is an empty ngOnInit()

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.ts [hunk 3]
```diff
@@ -72,7 +74,16 @@ export class SecurityComponent implements OnInit {
 
     removeTlsCaEntry(index: number) {
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.ts]: I'm trying to think of a way we can consolidate these functions. We have a lot of repetitive code. Maybe we could pass in the control name? Let me know if it's feasible or not.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.ts (hunk 2 above)
Comment (jooskim) [reply ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.ts]: There used to be something else under `// tmp` which obviously makes that line obsolete now. Will remove it as well as empty ngOnInit()

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.ts (hunk 3 above)
Comment (jooskim) [reply ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.ts]: Yup I just did consolidated those redundant functions into three which are now named as follows:

addNewFormArrayEntry(controlName: string);
createNewFormArrayEntry(controlName: string);
removeFormArrayEntry(controlName: string, index: number);

**Review by jak-atx — APPROVED** [review]

**Pull request merged**